### DISCUSSION TITLE
Questions regarding the qpdf functions.

### DISCUSSION THREAD

**Discussion by uddhavphatak** [qpdf/qpdf#1217] [Q&A] [answered]
hello,
I am having a problem with how the function getRoot() work. does it first find the object containing the "/root" and get the full object into dictionary? 
If it does this way, how does it look into the xref to find the Root?

Similarly, what type of Object is QPDFObjectHandle::parse("[/ICCBased ]"?

Thank you very much for your help

**Comment by m-holger** [comment, marked as answer]
The trailer of every PDF file contains a /Root entry with a reference to the document catalog. `getRoot()` returns this reference. 

> what type of Object is QPDFObjectHandle::parse("[/ICCBased ]")?

I am not sure what you are after. If you have `auto oh = QPDFObjectHandle::parse("[/ICCBased ]");` , then `oh` is a QPDFObjectHandle for which `oh.isArray()` returns true, `oh.getArrayNItems()` returns 1 and `oh.getArrayItem(0).isName()` returns true.

**Reply by uddhavphatak** [reply]
Thank you for your answer.

> I am not sure what you are after. If you have `auto oh = QPDFObjectHandle::parse("[/ICCBased ]");` , than `oh` is a QPDFObjectHandle for which `oh.isArray()` returns true, `oh.getArrayNItems()` returns 1 and `oh.getArrayItem(0).isName()` returns true.

yes, that was what I wanted to ask.

Thank you again.
